src/asmcnc/skavaUI/popup_info.py

```diff
@@ -1107,6 +1107,7 @@ def cancel_shutdown(*args):
             os.system('sudo shutdown -c')
 
         def shutdown_now(*args):
+            os.system('sudo sed -i "s/correct_shutdown=False/correct_shutdown=True/" config.txt')

```

Comment:
have one class that does the actual sudo calls for shutdown, restart and cancel. This function sets the flag to True.
find all instances where the console is reboot or shutdown and call the implemented functions instead.